server/embed/etcd.go

```diff
@@ -249,6 +260,14 @@ func StartEtcd(inCfg *Config) (e *Etcd, err error) {
 		return e, err
 	}
 
+	if e.cfg.ExperimentalEnabledTracing {
+		e.tctx = context.Background()
```

Comment:
I believe it is recommended not to do this: (fifth paragraph: https://golang.org/pkg/context/ )

It's probably better form to pass context directly into setupTracing. 

Also, why not invoke the defer tracingExporter.Shutdown right here (but before the conditional error return) on like 268?

Reply:
Because defer gets called straight away as the `StartEtcd` func returns. So I placed it in the `Close` where the `metricsListeners` is also shut down.

> // Close gracefully shuts down all servers/listeners.

But happy to adapt this differently if you have any other suggestions!

Will pass in context directly, wasn't too happy with that either you are right!

Reply:
Oh i didn't realize it was in a different function, the diff obscured that from me. I guess you'd have to wrap a lambda function for shutdown so you can pass in the context here, that should allow you to get away from storing context.

something like

```
   ctx := context.Background
   \\ tracing initialization
   e.shutdownTracing := function() {
     e.e.tracingExporter.Shutdown(ctx)
   }
```

Reply:
Makes sense, will do that. Was just following the current pattern of saving context in the struct that exists, but agreed on this approach!

Reply:
Done, please have a look thanks!